app/src/androidTest/java/au/com/shiftyjelly/pocketcasts/models/db/SearchHistoryDaoTest.kt

```diff
@@ -0,0 +1,274 @@
+    }
+
+    @Test
+    fun testInsertPodcastSearchHistory() {
+        val uuid = UUID.randomUUID().toString()
+        runTest {
+            searchHistoryDao.insert(createPodcastSearchHistoryItem(uuid))
+
+            assertTrue(searchHistoryDao.findAll().first().podcast?.uuid == uuid)
+        }
+    }
+
+    @Test
+    fun testInsertFolderSearchHistory() {
+        val uuid = UUID.randomUUID().toString()
+        runTest {
+            searchHistoryDao.insert(createFolderSearchHistoryItem(uuid))
+
+            assertTrue(searchHistoryDao.findAll().first().folder?.uuid == uuid)
+        }
+    }
+
+    @Test
+    fun testInsertEpisodeSearchHistory() {
+        val uuid = UUID.randomUUID().toString()
+        runTest {
+            searchHistoryDao.insert(createEpisodeSearchHistoryItem(uuid))
+
+            assertTrue(searchHistoryDao.findAll().first().episode?.uuid == uuid)
+        }
+    }
+
+    /* MULTIPLE INSERT OR REPLACE */
+    @Test
+    fun testMultipleInsertSameSearchTerms() {
+        runTest {
+            searchHistoryDao.insert(createTermSearchHistoryItem(SEARCH_TERM_TEST1))
+            val modifiedPrevious = searchHistoryDao.findAll().first().modified
+            searchHistoryDao.insert(createTermSearchHistoryItem(SEARCH_TERM_TEST1))
+
+            val result = searchHistoryDao.findAll()
+            assertEquals("Insert should replace, count should be 1", 1, result.size)
+            assertEquals(
+                "Replaced search term should be on top",
+                true,
+                result.first().modified > modifiedPrevious
+            )
+        }
+    }
+
+    @Test
+    fun testMultipleInsertUniqueSearchTerms() {
+        runTest {
+            searchHistoryDao.insert(createTermSearchHistoryItem(SEARCH_TERM_TEST1))
+            searchHistoryDao.insert(createTermSearchHistoryItem(SEARCH_TERM_TEST2))
+
+            val result = searchHistoryDao.findAll()
+            assertEquals("Unique search terms should be inserted, count should be 2", 2, result.size)
+            assertEquals(
+                "Last search term inserted should be on top",
+                SEARCH_TERM_TEST2,
+                result.first().term
+            )
+        }
+    }
+
+    @Test
+    fun testMultipleInsertSamePodcastSearchHistory() {
+        val uuid = UUID.randomUUID().toString()
+        runTest {
+            searchHistoryDao.insert(createPodcastSearchHistoryItem(uuid = uuid))
+            val modifiedPrevious = searchHistoryDao.findAll().first().modified
+            searchHistoryDao.insert(createPodcastSearchHistoryItem(uuid = uuid))
+
+            val result = searchHistoryDao.findAll()
+            assertEquals("Same podcast search insert should replace, count should be 1", 1, result.size)
+            assertEquals(
+                "Replaced podcast search history item should be on top",
+                true,
+                result.first().modified > modifiedPrevious
+            )
```

Comment:
Could just use `assertTrue`. Certainly not a big deal though.

Reply:
I had a different actual vs expected earlier that needed an equals assertion, `assertTrue` is better now, replaced in - 8f4ddd7